src/main/java/org/springframework/data/keyvalue/core/KeyValueTemplate.java

```diff
@@ -195,6 +195,7 @@ public <T> T update(Object id, T objectToUpdate) {
 		Assert.notNull(objectToUpdate, "Object to be updated must not be null");
 
 		String keyspace = resolveKeySpace(objectToUpdate.getClass());
+		Assert.notNull(keyspace, "Keyspace must not be null");
```

Comment:
should not be required.

Reply:
good catch -  added that one before the null check in `resolveKeySpace` - same goes for others